src/logic/safe/store/reducer/gatewayTransactions.ts

```diff
@@ -122,15 +136,30 @@ export const gatewayTransactionsReducer = handleActions<GatewayTransactionsState
         }
 
         const newTx = value.transaction
+
         if (label === 'queued') {
           if (newQueued?.[txNonce]) {
-            newQueued[txNonce] = [...newQueued[txNonce], newTx]
+            const txIndex = newQueued[txNonce].findIndex(({ id }) => sameString(id, newTx.id))
+
+            if (txIndex !== -1) {
```

Comment:
I suggest to change this to `txIndex >= 0` to keep it consistent with `ADD_HISTORY_TRANSACTIONS` above.